### DISCUSSION TITLE
Welcome to discussions about the Dioptra Platform!

### DISCUSSION THREAD

**Discussion by hbooth** [usnistgov/dioptra#140] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions to get started and install.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by RitikaVerma7** [comment]
Hi, 
This is Ritika Verma. 
Wanted to check if anyone else is also getting unhealthy container for dioptra and not able to run it on localhost? Can I get some help?

**Reply by keithmanville** [reply]
Hi Ritika,

The unhealthy container is due to a known broken health check (#475, #476). Despite this issue, Dioptra should work on localhost. Can you provide some additional information about your system (e.g. OS, docker version, etc.) and any specific error messages you received so we can help you debug?

**Reply by RitikaVerma7** [reply]
Sure. 
Docker version 27.1.1
OS: Ubuntu 22.4
Postgre container is always in restarting stage. Error - "/usr/local/bin/docker-entrypoint.sh: line 21: /etc/postgresql/postgres-passwd: No such file or directory"

Refer screenshots:

<img width="1920" alt="Screenshot 2024-08-12 at 7 38 29 PM" src="https://github.com/user-attachments/assets/302d347f-c3a3-472c-ac62-1149abea5fc5">
<img width="1920" alt="Screenshot 2024-08-12 at 7 38 37 PM" src="https://github.com/user-attachments/assets/01ba172b-c365-4878-a0fb-0b09614a03ae">

Thank you

**Reply by keithmanville** [reply]
Something likely went wrong when running `init-deployment.sh`. The postgres password is copied to a docker volume as part of that script.

Can you try shutting down your containers with `docker compose down` and the running the init deployment step again?

**Reply by RitikaVerma7** [reply]
Done.. Still the same. I tried to re-run from start as well.. I can try running on another machine if it will help.. 
Also, I am observing this error while running init - 

_exec /bin/bash: exec format error
Init Deployment: ERROR - Encountered an error when executing docker compose, exiting..._

**Reply by jkglasbrenner** [reply]
Looking over the screenshots more carefully, I'm noticing that you need to use `sudo` when you invoke docker, so that might be the issue here. The `init-deployment.sh` script does not have any sudo invocations inside of it, so the first time you come to a `docker compose` command things will fail.

Two things to try (either one or the other should work, no need to do both):

- Add your ubuntu user to the docker group: `sudo gpasswd -a ubuntu docker`. After this, running `docker images` should work and you shouldn't see the socket permission error. Note, you may need to logout and relogin for the group change to take effect.
- Alternatively, invoke the script with sudo privileges, `sudo ./init-deployment.sh`.

Before doing the above, I would purge any docker volumes just to be sure that things are clean: `docker volume prune -f --all`.

**Reply by RitikaVerma7** [reply]
Thank you so much.. I tried on another machine from scratch, it worked.

**Comment by RitikaVerma7** [comment]
Do we have any existing model, to experiment with or shall I create a custom model for running the test. I was able to download sample MNIST Data set.

**Reply by keithmanville** [reply]
Unfortunately we don't have any examples at the moment, as they were all deprecated with the release of v1. This is something we are working to address so it will be easier to get started with Dioptra in the future.

**Reply by RitikaVerma7** [reply]
Perfect! Thank you for all the assistance.
